src/components/Rectangle.jsx

```diff
@@ -3,32 +3,28 @@ import { Rect } from 'react-konva';
 import PropTypes from 'prop-types';
 
 export default class Rectangle extends Component {
-  state = {
-    x: this.props.x,
-    y: this.props.y,
-  }
-
-  handleDrag(event) {
-    this.setState({
-      x: event.target.attrs.x,
-      y: event.target.attrs.y,
-    });
-  }
-
   render() {
-    const { title, color } = this.props;
-    const { x, y } = this.state;
+    const {
+      title, color, x, y, updatePosition, id, handleTransform,
+    } = this.props;
+
     return (
       <Rect
         name={title}
         x={x}
         y={y}
+        id={id}
         width={50}
         height={50}
         stroke={color}
-        strokeWidth={2}
-        onClick={this.handleClick}
-        onDragEnd={event => this.handleDrag(event)}
+        strokeWidth={4}
+        strokeScaleEnabled={false}
+        onTransformEnd={event => handleTransform(id,
+          event.target.x(),
```

Comment:
Is there a way to destructure all you want from the `event` at once and then pass it to the handleTransform function instead of calling it severally.
Also, if the function is too large in here you can make a class function that takes in the event target and then passes it on to the handleTransform function.

Reply:
I will look into this right now.